mrp_production_partner_note/models/procurement_order.py

```diff
@@ -0,0 +1,19 @@
+# -*- encoding: utf-8 -*-
+##############################################################################
+# For copyright and license notices, see __openerp__.py file in root directory
+##############################################################################
+
+from openerp import models, api
+
+
+class ProcurementOrder(models.Model):
+    _inherit = 'procurement.order'
+
+    @api.model
+    def _prepare_mo_vals(self, procurement):
+        res = super(ProcurementOrder, self)._prepare_mo_vals(procurement)
+        sale_proc = procurement.move_dest_id.procurement_id
+        mrp_notes = sale_proc.sale_line_id.order_id.partner_id.mrp_notes
+        if mrp_notes:
+            res['notes'] = mrp_notes
```

Comment:
But there's no such possibility, because values can't be other type than string... but I agree it's more secure in general terms (but the length of the line is higher also, that's why in this case I preferred the other formula to fit in less lines).